tests/lib/test_Repo.py

```diff
@@ -6,16 +6,27 @@
 import pytest
 from onyo import Repo, OnyoInvalidRepoError
 
+def params(d):
+    return pytest.mark.parametrize(
+        argnames=(argnames := sorted({k for v in d.values() for k in v.keys()})),
+        argvalues=[[v.get(k) for k in argnames] for v in d.values()],
+        ids=d.keys(),
+    )
+
 
 #
 # instantiation
 #
-variants = {
-    'str': 'the-repo',
-    'Path': Path('the-repo'),
-}
-@pytest.mark.parametrize('variant', variants.values(), ids=variants.keys())
-def test_Repo_instantiate_types(tmp_path, variant):
+@params({
+        "Path": {"variant": Path("the-repo")},
```

Comment:
Nit: in later tests I notice you just indent these normally (with 4 spaces). I prefer that convention over this.